packages/material-ui/src/Stack/Stack.js

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  createUnarySpacing,
+  handleBreakpoints,
+  unstable_extendSxProp as extendSxProp,
+} from '@material-ui/system';
+import { deepmerge } from '@material-ui/utils';
+import { getValue } from '@material-ui/system/spacing';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+
+const getSideFromDirection = (direction) => {
+  switch (direction) {
+    case 'row':
+      return 'Left';
+    case 'row-reverse':
+      return 'Right';
+    case 'column':
+      return 'Top';
+    case 'column-reverse':
+      return 'Bottom';
+  }
+};
+
+const StackRoot = experimentalStyled(
+  'div',
+  {},
+  { name: 'Stack' },
+)(({ styleProps, theme }) => {
+  let styles = {
+    display: 'flex',
+    ...handleBreakpoints({ theme }, styleProps.direction, (propValue) => ({
+      flexDirection: propValue,
+    })),
+  };
+
+  if (styleProps.spacing) {
+    const transformer = createUnarySpacing(theme);
+    const styleFromPropValue = (propValue, breakpoint) => {
+      const direction = styles[breakpoint]?.flexDirection || styleProps.direction;
+      return {
+        '& > :not(styles) + :not(styles)': {
+          margin: 0,
+          [`margin${getSideFromDirection(direction)}`]: getValue(transformer, propValue),
+        },
+      };
+    };
+    styles = deepmerge(
+      styles,
+      handleBreakpoints({ theme }, styleProps.spacing, styleFromPropValue),
+    );
+  }
+
+  return styles;
+});
+
+const Stack = React.forwardRef(function Stack(inProps, ref) {
+  const themeProps = useThemeProps({ props: inProps, name: 'MuiStack' });
+  const props = extendSxProp(themeProps);
+  const { direction = 'column', spacing, divider, children, ...other } = props;
+  const lastChildIndex = React.Children.count(children) - 1;
+  const styleProps = {
+    direction,
+    spacing,
+  };
+
+  return (
+    <StackRoot styleProps={styleProps} ref={ref} {...other}>
+      {divider
+        ? React.Children.toArray(children).reduce(
```

Comment:
We would need to filter empty children:
```jsx
import * as React from "react";
import Divider from "@material-ui/core/Divider";
import Stack from "@material-ui/core/Stack";

export default function DividerStack() {
  return (
    <Stack
      direction="row"
      divider={<Divider orientation="vertical" />}
      spacing={2}
    >
      <div>1</div>
      {null}
      <div>1</div>
    </Stack>
  );
}
```

Reply:
Tried to solve this in 4abbfe4c0785abfb02da1773ed1bc0f50f0e8dc7

Reply:
Ah, yes, thanks for pulling that out! I added `Boolean` to the filter so it's a little more explicit what's going on, this is what we did at my old job to filter out null values.